edit.go

```diff
@@ -156,8 +157,36 @@ func (v *View) EditNewLine() {
 
 // MoveCursor mores the cursor relative from it's current possition
 func (v *View) MoveCursor(dx, dy int) {
+	newX, newY := v.cx+dx, v.cy+dy
+	// do nothing when no line exists
+	if len(v.lines) == 0 {
+		v.cx, v.cy = 0, 0
+		return
+	}
+	// If newY is more than all lines set it to the last line
+	if newY >= len(v.lines) {
+		newY = len(v.lines) - 1
+	}
+	if newY < 0 {
+		newY = 0
+	}
+	// if dx != to zero, then set dx to the next chr width
+	if dx != 0 {
+		line := v.lines[newY]
+		col := 0
+		for index, _ := range line {
+			col += runewidth.RuneWidth(line[index].chr)
+			if newX <= col && dx >= 0 {
+				dx = runewidth.RuneWidth(line[index].chr)
+				break
+			}
+			if newX < col && dx < 0 {
+				dx = -runewidth.RuneWidth(line[index].chr)
+				break
+			}
+		}
+	}
```

Comment:
I'm not sure if this part is correct. I feel like it kinda kills the option where user would like to use `dx` higher than 1.

Basically if user do something like this
```go
v.MoveCursor(4, 0)
```
it wouldn't move it by 4 characters, but instead it would move it by 1. It doesn't take into account the original `dx` value and override it with rune width.

Reply:
Another thing is, is it really necessary to count columns by adding rune width? The reason I ask is, there is a fix in `writeRune()` function which adds for each rune width higher than 1 another column in to the line. Meaning you can just check go the the index instead of trying to figure out the real column, because you already have that information in the line length kinda. 

Not sure if it's clear what I mean, but feel free to correct me. I just felt like this wouldn't be that necessary.

Reply:
For this one, the function of `moveCursor` didn't change. so you can still use `moveCursor` to move exactly what you enter. Also, for usage outside the package, maybe I could change `MoveCursor` to `MoveCursorWithRuneWidth` and make `MoveCursor` same as 'moveCursor'?

The fix in `writeRune` works well, if you enter a wide rune, it the move cursor correctly. 
However, if you have a width rune character already, it will not move cursor correct. For example "烫|aaa", if you press left arrow, it will still at the same position "烫|aaa".

Reply:
Oh yeah, `moveCursor` is ok, but that one is private. If you use this package in your application, you can only use the public version which is this one.

When using it in your application (for example when you need to overwrite default editor), and you would like to move cursor more than 1 character(rune) on X axis (left/right on line), it will always change `dx` to rune width only (which is in most cases 1).

To be honest, I don't think we need separate function because in my opinion `MoveCursor` should always move by runes, not by cells. Because you don't want to write in the middle of the rune. @mjarkk what do you think?

And with the fix in `writeRune`, this would be more consistent if the move would be by number of runes instead of cells.    
However, for that to work correctly the `dx` counting should be done differently. Something like this:
```go
	if dx != 0 {
		line := v.lines[newY]
		if dx < 0 {
			line = line[newX:v.cx] // not sure if the ranges are correct
		} else {
			line = line[v.cx:newX]
		}
		dx = 0
		for _, ch := range line {
			dx += runewidth.RuneWidth(ch.chr)
		}
	}
```

I guess the code needs to be fine-tuned to work correctly, but wanted to just show what I meant why the `dx` is not correct and how it could be fixed.

Reply:
fixed

Reply:
Doesn't really work :/ If you run `_examples/demo.go` you will see bunch of panics happening when scrolling in the text.